File: src/net/teamfps/java/serialization/ContainerType.java
/**
 * 
 */
package net.teamfps.java.serialization;

/**
 * @author Zekye
 *
 */
public enum ContainerType {

	UNKNOWN(0), FIELD(1), ARRAY(2), OBJECT(3);

	private final byte type;

	private ContainerType(int type) {
		this.type = (byte) type;
	}

	public byte getType() {
		return type;
	}

	public static ContainerType getType(byte type) {
		for (ContainerType t : values()) {
			if (t.getType() == type) {
				return t;
			}
		}
		assert (false);
		return UNKNOWN;
	}
}

File: src/net/teamfps/cplus/serialization/ContainerType.cpp
#include "ContainerType.h"
/**
* @author Zekye
*
*/
namespace teamfps {
	const ContainerType ContainerType::UNKNOWN = ContainerType(0);
	const ContainerType ContainerType::FIELD = ContainerType(1);
	const ContainerType ContainerType::ARRAY = ContainerType(2);
	const ContainerType ContainerType::OBJECT = ContainerType(3);

	ContainerType::ContainerType(ubyte type) : type(type) {}

	const ubyte ContainerType::getType() const {
		return type;
	}
}